File: backend/CNBCRssManager.API/Models/FeedItem.cs
namespace CNBCRssManager.API.Models
{
    public class FeedItem
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Link { get; set; }
        public string Description { get; set; }
        public DateTime PublishDate { get; set; }
        public bool IsRead { get; set; }
    }
}

File: frontend/src/components/FeedItem.js
import React from 'react';

const FeedItem = ({ item, onDelete, onMarkAsRead }) => (
  <div className={`feed-item ${item.isRead ? 'read' : 'unread'}`}>
    <h3>{item.title}</h3>
    <p className="description">{item.description}</p>
    <div className="item-footer">
      <span className="publish-date">Published: {new Date(item.publishDate).toLocaleString()}</span>
      <a href={item.link} target="_blank" rel="noopener noreferrer" className="read-more-link">
        Read More
      </a>
    </div>
    <div className="item-actions">
      {!item.isRead && (
        <button onClick={() => onMarkAsRead(item.id)} className="mark-read-btn">Mark as Read</button>
      )}
      <button onClick={() => onDelete(item.id)} className="delete-btn">Delete</button>
    </div>
  </div>
);

export default FeedItem;